fedimint-core/src/encoding/mod.rs

```diff
@@ -277,12 +360,16 @@ where
 
 impl<T> Decodable for Vec<T>
 where
-    T: Decodable,
+    T: Decodable + 'static,
 {
     fn consensus_decode<D: std::io::Read>(
         d: &mut D,
         modules: &ModuleDecoderRegistry,
     ) -> Result<Self, DecodeError> {
+        if TypeId::of::<T>() == TypeId::of::<u8>() {
+            // unsafe: we've just checked that T is `u8` so the transmute here is a no-op
+            return Ok(unsafe { mem::transmute::<Vec<u8>, Vec<T>>(consensus_decode_bytes(d)?) });
```

Comment:
wait you can't transmute `Vec` you have to go through `{to, from}_raw` methods

Reply:
see https://doc.rust-lang.org/nomicon/transmutes.html?highlight=Vec%20transmute#transmutes

Reply:
actually it is correct, because T is u8